tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/CombinationListingFeatureContext.php

```diff
@@ -72,88 +73,157 @@ private function fillCommand(UpdateCombinationFromListingCommand $command, array
     }
 
     /**
-     * @Then I should see following combinations of product :productReference in page :page limited to maximum :limit per page:
+     * @Then I should see following combinations list of product ":productReference":
      *
      * @param string $productReference
-     * @param int $page
-     * @param TableNode $dataTable
-     * @param int $limit
+     * @param TableNode $tableNode
      */
-    public function assertCombinationsPage(string $productReference, int $page, TableNode $dataTable, int $limit): void
+    public function assertCombinationsList(string $productReference, TableNode $tableNode): void
```

Comment:
I didn't review the whole file so I can't give much details for now (to give you early feedback)

But I think we should have two different steps:
- one with the whole combinations
- one with the pagination results

Of course they will probably share some common private functions